plugins/modules/iam_user.py

```diff
@@ -227,44 +228,27 @@ def convert_friendly_names_to_arns(connection, module, policy_names):
         module.fail_json(msg="Couldn't find policy: " + str(e))
 
 
-def wait_iam_exists(connection, module):
-    if module.check_mode:
-        return
-    if not module.params.get('wait'):
-        return
-
-    user_name = module.params.get('name')
-    wait_timeout = module.params.get('wait_timeout')
-
-    delay = min(wait_timeout, 5)
-    max_attempts = wait_timeout // delay
-
-    try:
-        waiter = connection.get_waiter('user_exists')
-        waiter.wait(
-            WaiterConfig={'Delay': delay, 'MaxAttempts': max_attempts},
-            UserName=user_name,
-        )
-    except botocore.exceptions.WaiterError as e:
-        module.fail_json_aws(e, msg='Timeout while waiting on IAM user creation')
-    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
-        module.fail_json_aws(e, msg='Failed while waiting on IAM user creation')
-
-
-def create_or_update_login_profile(connection, module):
+def create_or_update_login_profile(connection, module, name):
 
     # Apply new password / update password for the user
     user_params = dict()
-    user_params['UserName'] = module.params.get('name')
+    user_params['UserName'] = name
     user_params['Password'] = module.params.get('password')
     user_params['PasswordResetRequired'] = module.params.get('password_reset_required')
     retval = {}
 
     try:
-        retval = connection.update_login_profile(**user_params)
+        retval = connection.update_login_profile(aws_retry=True, **user_params)
+        if module.params.get('wait'):
+            # Wait for login profile to finish updating
+            connection.update_login_profile(aws_retry=True, **user_params)
```

Comment:
This does not actually wait, it just tries again.  In the case where we actually want to wait until changes have fully propagated on AWS' side we would need to do something like describe the login profile and see if it matches what we tried to update.

Reply:
While it doesn't really 'wait', it accomplishes the same thing by retrying the EntityTemporarilyUnmodifiable exception - there aren't any useful waiters, nor is there any sort of `status` returned, so I went with this approach

Reply:
Are we experiencing problems the module that would require us to wait?  Login profile should just be username and password so we definitely won't get credential information in a return response.  We only need to construct a waiter if the code is having problems as a result of not waiting for the AWS control plane to "catch up".  The purpose of this code is non-obvious and I'm not convinced it needs to be here.

Reply:
This doesn't honor the `wait_timeout`, though. Depending on how the dice are rolled during the jitter, this could end up being quite a bit longer than what the user specified.

Reply:
marking as resolved as it was determined that this method of waiting for updates isn't necessary